subsys/bluetooth/host/gatt.c

```diff
@@ -174,9 +174,15 @@ static ssize_t write_appearance(struct bt_conn *conn, const struct bt_gatt_attr
 }
 #endif /* CONFIG_BT_DEVICE_APPEARANCE_GATT_WRITABLE */
 
-#if CONFIG_BT_DEVICE_APPEARANCE_GATT_WRITABLE
+#if defined(CONFIG_BT_DEVICE_APPEARANCE_GATT_WRITABLE)
 	#define GAP_APPEARANCE_PROPS (BT_GATT_CHRC_READ | BT_GATT_CHRC_WRITE)
+#if defined(CONFIG_DEVICE_APPEARANCE_GATT_WRITABLE_AUTHEN)
 	#define GAP_APPEARANCE_PERMS (BT_GATT_PERM_READ | BT_GATT_PERM_WRITE_AUTHEN)
+#elif defined(CONFIG_BT_DEVICE_APPEARANCE_GATT_WRITABLE_ENCRYPT)
+	#define GAP_APPEARANCE_PERMS (BT_GATT_PERM_READ | BT_GATT_PERM_WRITE_ENCRYPT)
+#else
+	#define GAP_APPEARANCE_PERMS (BT_GATT_PERM_READ | BT_GATT_PERM_WRITE)
```

Comment:
It's a bit dangerous that the default is the most permissive? It is also not backwards-compatible. The default should be `BT_GATT_PERM_WRITE_AUTHEN`, like it was before.

How about we define a kconfig choice defaulting to 'authenticated'?